rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs

```diff
@@ -167,3 +170,290 @@ async fn make_an_attempt(
         })?
         .include(|ctx| interceptors.read_after_deserialization(ctx, cfg))
 }
+
+#[cfg(all(test, feature = "test-util"))]
+mod tests {
+    use super::invoke;
+    use crate::client::orchestrator::endpoints::StaticUriEndpointResolver;
+    use crate::client::retries::strategy::NeverRetryStrategy;
+    use crate::client::test_util::auth::{
+        http_auth_schemes_for_testing, identity_resolvers_for_testing, EmptyAuthOptionResolver,
+        EmptyAuthOptionResolverParams,
+    };
+    use crate::client::test_util::{
+        connector::OkConnector, deserializer::CannedResponseDeserializer,
+        endpoints::EmptyEndpointResolverParams, serializer::CannedRequestSerializer,
+        trace_probe::NoOpTraceProbe,
+    };
+    use aws_smithy_http::body::SdkBody;
+    use aws_smithy_runtime_api::client::interceptors::{
+        Interceptor, InterceptorContext, Interceptors,
+    };
+    use aws_smithy_runtime_api::client::orchestrator::ConfigBagAccessors;
+    use aws_smithy_runtime_api::client::runtime_plugin::{BoxError, RuntimePlugin, RuntimePlugins};
+    use aws_smithy_runtime_api::config_bag::ConfigBag;
+    use aws_smithy_runtime_api::type_erasure::TypeErasedBox;
+    use http::StatusCode;
+    use std::sync::Arc;
+    use tracing_test::traced_test;
+
+    fn new_request_serializer() -> CannedRequestSerializer {
+        CannedRequestSerializer::success(
+            http::Request::builder()
+                .body(SdkBody::empty())
+                .expect("request is valid"),
+        )
+    }
+
+    fn new_response_deserializer() -> CannedResponseDeserializer {
+        CannedResponseDeserializer::new(
+            http::Response::builder()
+                .status(StatusCode::OK)
+                .body(SdkBody::empty())
+                .map_err(|err| TypeErasedBox::new(Box::new(err)))
+                .map(|res| TypeErasedBox::new(Box::new(res))),
+        )
+    }
+
+    struct TestOperationRuntimePlugin;
+
+    impl RuntimePlugin for TestOperationRuntimePlugin {
+        fn configure(
+            &self,
+            cfg: &mut ConfigBag,
+            _interceptors: &mut Interceptors,
+        ) -> Result<(), BoxError> {
+            cfg.set_request_serializer(new_request_serializer());
+            cfg.set_response_deserializer(new_response_deserializer());
+            cfg.set_retry_strategy(NeverRetryStrategy::new());
+            cfg.set_endpoint_resolver(StaticUriEndpointResolver::http_localhost(8080));
+            cfg.set_endpoint_resolver_params(EmptyEndpointResolverParams::new());
+            cfg.set_auth_option_resolver_params(EmptyAuthOptionResolverParams::new());
```

Comment:
Can just use `StaticAuthOptionResolver` with an empty vec to remove these empty impls.